### PR TITLE
deploymentapi: Add CreateParams struct comments

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds new comments on `CreateParams` and `NewPayloadParams` which should
help inform our users of what each field does and if it's required or
not.

## Motivation and Context

Improve our Godoc.

## Types of Changes

- [x] Documentation

### CODE REVIEW

File: pkg/api/deploymentapi/create.go [hunk 1]
```diff
@@ -31,8 +31,16 @@ import (
 type CreateParams struct {
 	*api.API
 
+	// Request from which to create the deployment, by cominging this with
```
Comment (karencfv) [inline pkg/api/deploymentapi/create.go:34]:
```suggestion
	// Request from which to create the deployment, by combining this with
```
😄

File: pkg/api/deploymentapi/create.go [hunk 2]
```diff
@@ -31,8 +31,16 @@ import (
...
+	// the Overrides fields, certain fields can be overridden centrally through
+	// that struct.
+	//
+	// Additionally, a Request can be generated with depresourceapi.NewPayload()
+	// for more information on that struct, please refer to its godoc.
```
Comment (karencfv) [inline pkg/api/deploymentapi/create.go:39]:
```suggestion
	// Additionally, a Request can be generated with depresourceapi.NewPayload().
	// For more information on that struct, please refer to its godoc.
```

**Review by karencfv — APPROVED** [review]
Couple of typos, but thank you for adding these! Very nice 💅

**Comment by marclop** [comment]
I'm the king of the typos! 👑

**Review by karencfv — APPROVED** [review]

**Pull request merged**